eggtimer.js

```diff
@@ -1,326 +1,681 @@
+            const labelsUpdated = results[1];
+            if (state === "closed" && labelsUpdated)
+                return Promise.resolve(true);
+            return Promise.reject("cleanup failed");
+        })
+        .then((result) => {
+             assert(result === true);
+             processNextPR();
+             return Promise.resolve(true);
+         })
+        .catch((err) => {
+            if (errorLabel === null) {
+                return Promise.resolve(true);
+            } else {
+               console.error("Error merging auto_branch(" + prContext.autoSha + ") into master:", err);
+               processNextPR();
+               let params = commonParams();
+               params.number = prContext.pr.number.toString();
+               params.labels = [];
+               params.labels.push(errorLabel);
+               return addLabels(params);
+            }
+        })
+        .then((result) => {
+             assert(result === true);
+        })
+        .catch((err) => {
+            console.error("Error setting label " + errorLabel, err);
+        });
 }
 
-function isApproved(obj) {
-    if (!('reviews' in obj)) {
-        return false;
-    }
-    else if (Object.keys(obj.reviews).length <= 0) {
-        return false;
-    }
-    for (var id in obj.reviews) {
-        if (!obj.reviews[id]) {
-            return false;
-        }
-    }
-    return true;
+function startMerging(prContext) {
+    console.log("Start merging PR" + prContext.pr.number, prContext.pr.head.sha);
+    let getParams = commonParams();
+    getParams.ref = "heads/master";
+    let masterSha = null;
+    getReference(getParams)
+        .then((sha) => {
+            masterSha = sha;
+            let params = commonParams();
+            params.ref = "pull/" + prContext.pr.number.toString() + "/merge";
+            return getReference(params);
+        })
+        .then((sha) => {
+            let params = commonParams();
+            params.sha = sha;
+            return getCommit(params);
+        })
+        .then((obj) => {
+            let params = commonParams();
+            params.tree = obj.treeSha;
+            params.message = prContext.pr.title + endOfLine + prContext.pr.body + endOfLine + "(PR #" + prContext.pr.number.toString() + ")";
+            params.parents = [];
+            params.parents.push(masterSha.toString());
+            return createCommit(params);
+        })
+        .then((sha) => {
+            let params = commonParams();
+            params.body = "Merging PR #" + prContext.pr.number.toString();
+            params.sha = sha;
+
+            let tagParams = commonParams();
+            tagParams.sha = sha;
+            let tagPromise = null;
+            if (prContext.tagSha === null) {
+                tagParams.ref = "refs/" + mergingTag(prContext.pr.number);
+                tagPromise = createReference(tagParams);
+            } else {
+                tagParams.ref = mergingTag(prContext.pr.number);
```

Comment:
How would a mergingTag() caller know whether to prefix the result with "refs/" (like a few lines above) or not (like on this line)? Please fix the mergingTag() API.